src/ucp/core/ucp_ep.c

```diff
@@ -331,9 +327,14 @@ ucs_status_t ucp_worker_create_ep(ucp_worker_h worker, unsigned ep_init_flags,
         goto err_destroy_ep_base;
     }
 
-    ucs_list_add_tail(&worker->all_eps, &ucp_ep_ext_gen(ep)->ep_list);
+    if (ep_init_flags & UCP_EP_INIT_FLAG_INTERNAL) {
+        ucp_ep_flush_state_reset(ep);
```

Comment:
to call `ucp_ep_disconnected` for all EPs regardless of their type

Reply:
maybe disconnected should not assume flush_state for internal ep?

Reply:
it assumes fkush_state, since it does flush for reply EP

Reply:
why need it for internal ep and not for any other ep?

Reply:
> why need it for internal ep and not for any other ep?

other EPs reset their flush state in CM-related code (conn_to_sock_addr) or after connection matching is no longer needed (conn_to_worker_addr).

Reply:
what happens if we try to destroy/flush other ep before it's connected?

Reply:
it is controlled here:
https://github.com/openucx/ucx/blob/fc0f2557534a7a35c29f88ea7b2b7c70e087c5a4/src/ucp/core/ucp_ep.c#L2770

flush_state is always valid for EP created using CM

Reply:
would it help to check INTERNAL flag in there?

Reply:
> would it help to check INTERNAL flag in there?

I think, no.
We should flush reply EP which is INTERNAL one

Reply:
so, i think need to move flush state reset to the place where we create the reply ep
so it will be the same logical layer as other eps (w.g when they are created)